### DISCUSSION TITLE
Mag Safe Charging Stopped Working

### DISCUSSION THREAD

**Discussion by Juliannnnnnnnnnnnnnnnnnn** [AppHouseKitchen/AlDente-Battery_Care_and_Monitoring#1043] [Q&A] [answered]
Today, the magsafe charger suddenly stopped working. The light is off, and when I plug the adapter into the wall, the computer does not recognize that it is plugged in.

However, I can use the same adapter and charge the computer via the USB port. The magsafe cable is also in perfect shape. So the problem seems to be on the the computer side.

I'm using a MacBook Air 2022 M2 on Ventura 13.5.2. Updating the operating system did not help, nor did uninstalling or reinstalling AlDente, nor did restarting, nor did shutting down and only restarting after two minutes.

Please help!

**Comment by MatthiasKerbl** [comment]
Hi @Juliannnnnnnnnnnnnnnnnnn,

did you try the MagSafe cable on a different MacBook?
Could be a hardware defect of the MagSafe cable or the MagSafe plug of the MacBook.

Best,
Matthias

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply]
Thanks for your quick reply. I cannot test the cable itself (as opposed to the adapter) since I don't have another macbook that uses that format. But the adapter works. Since the whole macbook is only about 2 months old, it would be surprising if the cable were broken (and it shows not physical defects); same for the plug of the MacBook itself. Is there any way to make sure I've completely undone whatever AlDente does with the settings of the computer and reset things to factory settings (without, ideally, erasing the whole computer and starting over)?

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply]
PS I really do appreciate that you got right back to me!

**Reply by MatthiasKerbl** [reply]
Yes, to fully erase AlDente from your system, just follow the [uninstallation guide](https://apphousekitchen.com/how-to-uninstall-aldente-pro/) on our blog and restart your MacBook.

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply]
Unfortunately, this did not solve the problem.

**Reply by MatthiasKerbl** [reply]
In general, it is highly unlikely that this issue has anything to do with AlDente. For me, it seems like some kind of hardware defect. I would contact Apple support.

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply]
Thanks for your continued support.

I ran diagnostics and get PPP020 - No Power Adapter detected. 

How can I be certain this has nothing to do with AlDente? It's just strange that this would happen in a new computer.

**Reply by metebyte** [reply]
Funnily enough, they are always very sure about this issue.

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply, marked as answer]
To be fair, I took it to the mac store, and the magsafe cable was defective...

**Comment by DamonHD** [comment]
I have occasionally found dirt to obstruct the MagSafe contacts, or get on the body and stop the contacts touching, and stop it working totally or in part.  I may be worth carefully cleaning the plug and the socket.

Rgds

Damon

**Reply by Juliannnnnnnnnnnnnnnnnnn** [reply]
Thanks! I've tried this, but I will try again.
